org.eclipse.xtext/src/org/eclipse/xtext/mwe/RuntimeResourceSetInitializer.java

```diff
@@ -37,32 +37,21 @@ public class RuntimeResourceSetInitializer {
 	private IResourceServiceProvider.Registry registry;
 
 	public List<String> getClassPathEntries() {
-		List<String> pathes = Lists.newArrayList();
+		List<String> paths = new ArrayList<>();
 		String classPath = System.getProperty("java.class.path");
 		String separator = System.getProperty("path.separator");
 		String[] strings = classPath.split(separator);
-		for (String path : strings) {
-			pathes.add(path);
-		}
-		return pathes;
+		Collections.addAll(paths, strings);
+		return paths;
 	}
 
 	protected Multimap<String, URI> getPathToUriMap(List<String> pathes) {
 		return getPathToUriMap(pathes, null);
 	}
 	
 	protected Multimap<String, URI> getPathToUriMap(List<String> pathes, final UriFilter filter) {
-		return traverser.resolvePathes(pathes, new Predicate<URI>() {
-			@Override
-			public boolean apply(URI input) {
-				boolean result = true;
-				if (filter != null)
-					result = filter.matches(input);
-				if (result)
-					result = registry.getResourceServiceProvider(input) != null;
-				return result;
-			}
-		});
+		return traverser.resolvePathes(pathes,
+				input -> filter.matches(input) && registry.getResourceServiceProvider(input) != null);
```

Comment:
Previously `filter` was only used if it wasn't null.

Reply:
Right. I restored that behavior.
Since it isn't as simple as before then, I can also restore the previous code if you prefer.

Reply:
The change is fine.